### DISCUSSION TITLE
Have `file.modpath()` run the OCI url when called from OCI

### DISCUSSION THREAD

**Discussion by obeleh** [kcl-lang/kcl#1800] [Ideas]
I was discussing `file.modpath()` with @zong-zhe [here](https://cloud-native.slack.com/archives/C05TC96NWN8/p1735117844392429?thread_ts=1734721482.574549&cid=C05TC96NWN8)

I was wondering what it would return as I was trying to rely on it's behavior in a module: [example](https://github.com/obeleh/kcl-optional-options/tree/modpath?tab=readme-ov-file#modpath)

On one hand I want the behavior of a module not to change when I release it to an OCI registry. On the other hand, I don't want it to "lie" either. So I think the modpath for `oci://something.com/a` should be `"oci://something.com/a"` 

Things that I think would complicate it:
- If one was trying to find out the "parent directory". I think it should return `"oci://something.com"` but getting the parent directory of `"oci://something.com"` would be equivalent to getting the parent directory of `/`

**Comment by Peefy** [comment]
cc @zong-zhe @He1pa
